### PR TITLE
Feature/contract negotiation role

### PR DESCRIPTION
**Pull request by davidjovanovic**

### CODE REVIEW

File: negotiation/src/main/java/it/eng/negotiation/model/Agreement.java [hunk 1]
```diff
@@ -30,11 +30,11 @@
 @Document(collection = "agreements")
 public class Agreement {
 
-	@JsonIgnore
-	private String consumerPid;
-	
-	@JsonIgnore
-	private String providerPid;
+//	@JsonIgnore
```
Comment (IgorBalog-Eng) [inline negotiation/src/main/java/it/eng/negotiation/model/Agreement.java:33]: If not needed we should remove it from code.
In Offer it is removed, not left commented

File: negotiation/src/main/java/it/eng/negotiation/rest/api/ContractNegotiationAPIController.java [hunk 2]
```diff
@@ -113,7 +113,7 @@ public ResponseEntity<GenericApiResponse<JsonNode>> sendAgreement(@RequestBody J
 	@PutMapping(path = "/{contractNegotiationId}/approve")
     public ResponseEntity<GenericApiResponse<JsonNode>> approveContractNegotiation(@PathVariable String contractNegotiationId) {
         log.info("Handling contract negotiation approved");
-        ContractNegotiation contractNegotiationApproved = apiService.handleContractNegotiationAgreed(contractNegotiationId);
+        ContractNegotiation contractNegotiationApproved = apiService.approveContractNegotiation(contractNegotiationId);
```
Comment (IgorBalog-Eng) [inline negotiation/src/main/java/it/eng/negotiation/rest/api/ContractNegotiationAPIController.java:116]: Nice rename

File: negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationAPIService.java [hunk 3]
```diff
@@ -306,10 +319,22 @@ public ContractNegotiation handleContractNegotiationAgreed(String contractNegoti
 				credentialUtils.getConnectorCredentials());
 		if(response.isSuccess()) {
 			log.info("Updating status for negotiation {} to agreed", contractNegotiation.getId());
-			ContractNegotiation contractNegtiationAgreed = contractNegotiation.withNewContractNegotiationState(ContractNegotiationState.AGREED);
-			contractNegotiationRepository.save(contractNegtiationAgreed);
 			log.info("Saving agreement..." + agreementMessage.getAgreement().getId());
 			agreementRepository.save(agreementMessage.getAgreement());
+			
+			ContractNegotiation contractNegtiationAgreed = ContractNegotiation.Builder.newInstance()
```
Comment (IgorBalog-Eng) [inline negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationAPIService.java:325]: Can this one be changed with withNewContractNegotiationState(ContractNegotiationState.AGREED)?

File: negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationConsumerService.java [hunk 4]
```diff
@@ -105,7 +107,9 @@ public void handleAgreement(ContractAgreementMessage contractAgreementMessage) {
     			.callbackAddress(contractAgreementMessage.getCallbackAddress())
     			.assigner(contractNegotiation.getAssigner())
     			.state(ContractNegotiationState.AGREED)
+    			.role(contractNegotiation.getRole())
```
Comment (IgorBalog-Eng) [inline negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationConsumerService.java:110]: Can this one be changed with withNewContractNegotiationState(ContractNegotiationState.AGREED)?

**Review by IgorBalog-Eng — COMMENTED** [review]
PR review

File: negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationAPIService.java (hunk 3 above)
Comment (IgorBalog-Eng) [reply negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationAPIService.java:325]: I just verified in the changed code, there is agreement added from AgreementMessage, not previous contract negotiation

File: negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationConsumerService.java (hunk 4 above)
Comment (IgorBalog-Eng) [reply negotiation/src/main/java/it/eng/negotiation/service/ContractNegotiationConsumerService.java:110]: Similar like previous
agreement(contractAgreementMessage.getAgreement())
not from contractNegotiation

**Review by IgorBalog-Eng — APPROVED** [review]

File: negotiation/src/main/java/it/eng/negotiation/model/Agreement.java (hunk 1 above)
Comment (davidjovanovic) [reply negotiation/src/main/java/it/eng/negotiation/model/Agreement.java:33]: No problemo

**Pull request merged**